pkg/api/types.go

```diff
@@ -173,9 +173,9 @@ type VolumeSource struct {
 	HostPath *HostPathVolumeSource `json:"hostPath"`
 	// EmptyDir represents a temporary directory that shares a pod's lifetime.
 	EmptyDir *EmptyDirVolumeSource `json:"emptyDir"`
-	// GCEPersistentDisk represents a GCE Disk resource that is attached to a
+	// CloudPersistentDisk represents a cloud-provided storage resource that is attached to a
 	// kubelet's host machine and then exposed to the pod.
-	GCEPersistentDisk *GCEPersistentDiskVolumeSource `json:"persistentDisk"`
+	CloudPersistentDisk *CloudPersistentDiskVolumeSource `json:"persistentDisk"`
```

Comment:
I don't think this is going to work.  You'll overload this struct for every type of cloud provider, which means you have to type it somehow.  Some properties would be relevant to some but not all of the providers.  

Better, I think, to have an explicit struct per type.

We agree there will be commonalities between them.  Good code structure can promote reuse for the common pieces.